src/modules/account/account.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import Account from '../../entity/account.entity';
+import { Repository } from 'typeorm';
+import { CreateAccountDTO } from './account.dto';
+@Injectable()
+export class AccountService {
+	constructor(
+		@InjectRepository(Account)
+		private readonly accountRepository: Repository<Account>,
+	) {}
+
+	async create(dto: CreateAccountDTO) {
+		const account = this.accountRepository.create(dto);
+
+		await this.accountRepository.save(account);
+	}
+
+	async findByIdAndType(socialId: string, type: string) {
+		const account = await this.accountRepository.findOne({
+			where: {
+				socialId,
+				type,
+			},
+			relations: { user: true },
+		});
+
+		return account;
```

Comment:
2 cách viết này khác nhau gì em?
```suggestion
		const account = this.accountRepository.findOne({
			where: {
				socialId,
				type,
			},
			relations: { user: true },
		});

		return account;
```

Reply:
Khi viết theo cách của anh suggest thì nó sẽ không chỉ cần await 1 lần khi mà mình sử dụng hàm đó ạ. Đối với cách của em viết thì phải await 2 lần ạ. Mặc dù kết quả vẫn giống nhau.

Reply:
Nhưng có khác biệt nào lớn giữa 2 cái này không?
```js 
function get(){
return await this.accountRepository.findOne()
}
// vs
function get(){
return this.accountRepository.findOne()
}

async function main(){
await get();
}

Reply:
Đối với hàm get thứ nhất sẽ chờ promise được giải quyết rồi mới trả về kết quả. Còn hàm get thứ 2 sẽ trả về promise ngay lập tức.